pkg/apicast/apicast.go

```diff
@@ -169,16 +161,15 @@ func (a *APIcast) deploymentEnv() []v1.EnvVar {
 	return env
 }
 
-func (a *APIcast) Deployment() *appsv1.Deployment {
-
+func (a *APIcast) Deployment() (*appsv1.Deployment, error) {
```

Comment:
We allow returning errors in the returning of objects model?

This seems a conceptual change on how we handle the data model objects.
Are we going to allow errors on it from now on? If so, this change applies to 3scale-operator too and its models?

Reply:
The only source of error is the method `controllerutil.SetControllerReference` because it checks if the objects already has a owner. If it already has and it is different to the one trying to be set, error is raised. 

Assuming the objects do not have owners, because they are being created, it is safe to assign owner the way it was before:
```
o.SetOwnerReferences(append(o.GetOwnerReferences(), owner))
```
I will change it

Reply:
I remember the motivation behind doing this way: I did not want to implement this function:
```
// asOwner returns an owner reference set as the tenant CR         
func asOwner(a *appsv1alpha1.APIcast) metav1.OwnerReference {      
    trueVar := true                                                
    return metav1.OwnerReference{                                  
        APIVersion: appsv1alpha1.SchemeGroupVersion.String(),      
        Kind:       appscommon.APIcastKind,                        
        Name:       a.Name,                                        
        UID:        a.UID,                                         
        Controller: &trueVar,                                      
    }                                                              
}                                                                  
```
This OwnerReference object building is done by `controllerutil.SetControllerReference` method and I wanted to use it instead of implementing it. The toll was having errors returned. If you prefer implementing it, no problem.

Reply:
changed

Reply:
Just to clarify, I was not against changing it if you prefer the new way you implemented it.
I was just asking whether we changed the way of dealing with it (which might be the appropriate one).
I see you've changed it already but if you prefer the previous way I'm ok with it too if it means we in general would be using the new approach